### PR TITLE
[Remove] LegacyESVersion.V_7_4_* and V_7_5_* constants

### PR DESCRIPTION
**Pull request by nknize**
Removes all usages of LegacyESVersion.V_7_4_ and LegacyESVersion.V_7_5 version constants
along with ancient API logic.

relates #2747
depends on #4703

### CODE REVIEW

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Needs a rebase, thx.

**Comment by nknize** [comment]
jenkins heap space took a dump; refiring

**Comment by nknize** [comment]
Another #4162 refiring

File: server/src/main/java/org/opensearch/indices/recovery/RecoverySourceHandler.java [hunk 1]
```diff
@@ -720,7 +720,7 @@ void createRetentionLease(final long startingSeqNo, ActionListener<RetentionLeas
                 // it's possible that the primary has no retention lease yet if we are doing a rolling upgrade from a version before
                 // 7.4, and in that case we just create a lease using the local checkpoint of the safe commit which we're using for
                 // recovery as a conservative estimate for the global checkpoint.
-                assert shard.indexSettings().getIndexVersionCreated().before(LegacyESVersion.V_7_4_0)
+                assert shard.indexSettings().getIndexVersionCreated().before(LegacyESVersion.fromId(7040099))
```
Comment (reta) [inline server/src/main/java/org/opensearch/indices/recovery/RecoverySourceHandler.java:723]: Do we actually need this assertion `shard.indexSettings().getIndexVersionCreated().before(LegacyESVersion.fromId(7040099))`? The 3.0.0 clustrer will never work with 7.x anyway, right?

Comment (nknize) [reply server/src/main/java/org/opensearch/indices/recovery/RecoverySourceHandler.java]: No... this can be removed. Thx!

**Review by reta — APPROVED** [review]

**Review by dreamer-89 — APPROVED** [review]

**Pull request merged**